.github/workflows/test-linux-gpu.yml

```diff
@@ -0,0 +1,101 @@
+name: Test (Linux/GPU)
+
+on:
+  workflow_dispatch:
+    inputs:
+      build-type:
+        description: Build in Release or Debug?
+        required: true
+        default: Debug
+        type: choice
+        options:
+          - Debug
+          - Release
+  workflow_call:
+    inputs:
+      build-type:
+        required: true
+        default: Debug
+        type: string
+
+jobs:
+  #
+  # Build shards for linux
+  #
+  Linux:
+    name: Build (${{ github.event.inputs.build-type || inputs.build-type }})
+    runs-on: self-hosted
+    outputs:
+      run-tests: ${{ steps.setup.outputs.run-tests }}
+      run-extra-tests: ${{ steps.setup.outputs.run-extra-tests }}
+    steps:
+      - name: Setup
+        id: setup
+        run: |
+          echo "::set-output name=build-type::${{ github.event.inputs.build-type || inputs.build-type }}"
+      - name: Checkout shards
+        uses: actions/checkout@v3
+        with:
+          repository: fragcolor-xyz/shards
+          fetch-depth: 1
+          submodules: recursive
+      - name: Set up dependencies
+        run: |
+          ./bootstrap
+      - name: Build
+        run: |
+          cmake -Bbuild -G Ninja -DCODE_COVERAGE=1 -DCMAKE_BUILD_TYPE=${{ steps.setup.outputs.build-type }}
+          cmake --build build --target shards test_gfx
```

Comment:
Not using `ninja shards`?

Reply:
I prefer this form since it's agnostic to the generator being used